--- createrepo-0.9.8/createrepo/__init__.py~	2009-08-29 14:48:13.000000000 +0300
+++ createrepo-0.9.8/createrepo/__init__.py	2009-08-29 14:49:26.166771314 +0300
@@ -468,7 +468,7 @@
            
         if po.checksum in (None, ""):
             raise MDError, "No Package ID found for package %s, not going to" \
-                           " add it" % po
+                           " add it" % rpmfile
         
         return po
 
